fuel-core/src/service/modules.rs

```diff
@@ -292,12 +295,16 @@ impl BlockProducerRelayer for MaybeRelayerAdapter {
             if let Some(sync) = self.relayer_synced.as_ref() {
                 sync.await_synced().await?;
             }
-        }
 
-        Ok(self
-            .database
-            .get_finalized_da_height()
-            .await
-            .unwrap_or_default())
+            Ok(self
+                .database
+                .get_finalized_da_height()
+                .await
+                .unwrap_or_default())
+        }
+        #[cfg(not(feature = "relayer"))]
```

Comment:
Here I added a new logic(to return a default value)=( Because the `RelayerDB` trait now lives inside of `fuel-relayer`(instead of `fuel-core-interfaces`) I can't use the `get_finalized_da_height` if `relayer` feature is disabled.